src/test/samplebuffertest.cpp

```diff
@@ -3,50 +3,50 @@
 #include <QtDebug>
 
 #include "test/mixxxtest.h"
-#include "util/circularsamplebuffer.h"
+#include "util/readaheadsamplebuffer.h"
 
-class CircularSampleBufferTest: public MixxxTest {
+class ReadAheadSampleBufferTest: public MixxxTest {
 public:
-    CircularSampleBufferTest()
+    ReadAheadSampleBufferTest()
         : m_writeValue(CSAMPLE_ZERO),
           m_readValue(CSAMPLE_ZERO) {
     }
 
 protected:
     static const SINT kCapacity;
 
-    SINT writeTail(CircularSampleBuffer* pSampleBuffer, SINT size) {
-        const SampleBuffer::WritableChunk writableChunk(
-                pSampleBuffer->writeToTail(size));
-        for (SINT i = 0; i < writableChunk.size(); ++i) {
-            writableChunk[i] = m_writeValue;
+    SINT write(mixxx::ReadAheadSampleBuffer* pSampleBuffer, SINT size) {
+        const mixxx::SampleBuffer::WritableSlice writableSlice(
+                pSampleBuffer->write(size));
+        for (SINT i = 0; i < writableSlice.size(); ++i) {
+            writableSlice[i] = m_writeValue;
             m_writeValue += CSAMPLE_ONE;
         }
-        return writableChunk.size();
+        return writableSlice.size();
     }
 
-    SINT readHeadAndVerify(CircularSampleBuffer* pSampleBuffer, SINT size) {
-        const SampleBuffer::ReadableChunk readableChunk(
-                pSampleBuffer->readFromHead(size));
-        for (SINT i = 0; i < readableChunk.size(); ++i) {
-            EXPECT_EQ(readableChunk[i], m_readValue);
+    SINT readFifoAndVerify(mixxx::ReadAheadSampleBuffer* pSampleBuffer, SINT size) {
+        const mixxx::SampleBuffer::ReadableSlice readableSlice(
+                pSampleBuffer->readFifo(size));
```

Comment:
Here it looks like you get mixed up yourselves. I think Fifo is right here if you swap the implementation.